### PR TITLE
#423 added appdynamics again including javascript agent

### PR DESCRIPTION
**Pull request by ChristianLohmann**

### CODE REVIEW

File: server/src/monitoring/appdynamics.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+;/* Version 6ed9e8be6594938a3179b249648f60c0 v:4.1.8.7, c:633a5cf8179243e14906a663132c69e8598f7c88, b:13635 n:52-4.1.8.next-controller */(function(){(function(){if(!window.ADRUM&&!0!==window["adrum-disable"]){var g=window.ADRUM={};window["adrum-start-time"]=window["adrum-start-time"]||(new Date).getTime();(function(a){(function(a){a.tc=function(){for(var a=[],c=0;c<arguments.length;c++)a[c-0]=arguments[c];for(c=0;c<a.length;c++){var b=a[c];b&&b.H()}}})(a.monitor||(a.monitor={}))})(g||(g={}));(function(a){a=a.conf||(a.conf={});a.beaconUrlHttp="http://col-eum.monitoring-appdynamics.zalan.do:80";a.beaconUrlHttps="https://col-eum.monitoring-appdynamics.zalan.do:443";a.corsEndpointPath="/eumcollector/beacons/browser/v1";
```
Comment (prayerslayer) [inline server/src/monitoring/appdynamics.js]: These `appdynamics.zalan.do` URLs look like they should be set via a configuration option.

File: server/src/monitoring/monitoring.js [hunk 2]
```diff
@@ -1,15 +1,44 @@
...
+        xmlFile = String(fs.readFileSync('/agents/appdynamics-jvm/conf/controller-info.xml'));
+    } catch (err) {
+        winston.error('Could not read appdynamics config XML.', err.message);
+    }
+    xml2js(xmlFile, function (err, result) {
+        if (err) {
+            winston.error('Could not parse appdynamics config XML. Error: %s. Content: %s.', err.message, xmlFile);
+            return;
+        }
+        var config = Object
```
Comment (prayerslayer) [inline server/src/monitoring/monitoring.js]: Last time I had to do funky things like checking if a string looked like a number and parsing it, if it did. Not anymore?

File: server/src/yourturn.js [hunk 3]
```diff
@@ -16,13 +19,11 @@ require('./monitoring/monitoring');
 require('./env');
 
 // this is the actual server code
-var fs = require('fs'),
-    express = require('express'),
+var express = require('express'),
```
Comment (prayerslayer) [inline server/src/yourturn.js]: What happened here?

Comment (ChristianLohmann) [inline server/src/yourturn.js]: I first added appdynamics config to yourturn.js. Therefore i had to move fs require to the top. I then realized the monitoring.js which encapsulates any monitoring config and i moved it over there.

File: server/src/monitoring/monitoring.js (hunk 2 above)
Comment (ChristianLohmann) [inline server/src/monitoring/monitoring.js]: i didn't change here anything on that xml -> json reading part. so what do you mean exactly?

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,5 @@
-FROM registry.opensource.zalan.do/stups/node:4.4-20
+# FROM registry.opensource.zalan.do/stups/node:5.10-22
```
Comment (prayerslayer) [inline Dockerfile]: Use `5.8-22`

**Comment by ChristianLohmann** [comment]
👍

File: server/src/yourturn.js (hunk 3 above)
Comment (prayerslayer) [inline server/src/yourturn.js]: These changes seem random, can we revert them?

**Comment by prayerslayer** [comment]
👍

**Pull request merged**